tests/test_dmlscheduler.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+import time
+import tests.context
+
+import logging
+import numpy as np
+from custom.keras               import get_optimizer
+from core.runner                import DMLRunner
+from core.runner_actor          import DMLRunnerActor
+from models.keras_perceptron    import KerasPerceptron
+from core.utils.dmljob          import DMLJob, serialize_job, deserialize_job
+from core.scheduler             import DMLScheduler
+from examples.labelers          import mnist_labeler
+from queue import Queue
+print("testing")
+
+
+import ray
+
+
+def make_dataset_path():
+    return 'datasets/mnist'
+
+
+def make_config():
+    return {}
+
+
+def make_model_json():
+    m = KerasPerceptron(is_training=True)
+    model_architecture = m.model.to_json()
+    model_optimizer = get_optimizer(m.model)
+    model_json = {
+        "architecture": model_architecture,
+        "optimizer": model_optimizer
+    }
+    return model_json
+
+
+def make_initialize_job(model_json):
+    initialize_job = DMLJob(
+        "initialize",
+        model_json,
+        "keras"
+    )
+    return initialize_job
+
+scheduler = DMLScheduler()
+
+#Check that the scheduling/running functionality is maintained.
+def test_dmlscheduler_sanity():
+    model_json = make_model_json()
+    initialize_job = make_initialize_job(model_json)
+    scheduler.add_job(initialize_job)
+    scheduler._runners_run_next_jobs()
+    start = time.time()
+    while not scheduler.processed:
+        scheduler._runners_run_next_jobs()
+    initial_weights = scheduler.processed.pop(0)
+    print(time.time()-start)
+    assert type(initial_weights) == list
+    assert type(initial_weights[0]) == np.ndarray
+    scheduler.reset()
+
+#Check that running 10 jobs with Ray takes at most half the 
+# time to run them serially (i.e. there is a speedup)
+def test_dmlscheduler_speedup_naive():
```

Comment:
I don't think this is a good test. It's a good proof to justify why this scheduler is better, but I don't think it's something that should be run every time. It also depends on your hardware resources. I would remove this.

Reply:
I completely agree with your logic, and I actually thought about this problem for a while. Even though the test is naive (as I mentioned before), I still think it's necessary to have it. My reasoning is that if someone in the future messes with the class and accidentally breaks the threading and not the core functionality (i.e. every runner runs the same job), then we won't know since the tests would still pass.  I can always just lower the bar of the test if you're afraid of varying hardware resources causing occasional fails. 

If you have faith that the situation I described won't occur, though, I'll go ahead and delete the test.